src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1773,11 +1773,16 @@ Base.vcat(dfs::AbstractDataFrame...;
            cols::Union{Symbol, AbstractVector{Symbol},
                        AbstractVector{<:AbstractString}}=:setequal,
            source::Union{Nothing, Symbol, AbstractString,
-                         Pair{<:Union{Symbol, AbstractString}, <:AbstractVector}}=nothing)
+                         Pair{<:Union{Symbol, AbstractString}, <:AbstractVector}}=nothing,
+           init::AbstractDataFrame=DataFrame())
 
 Efficiently reduce the given vector or tuple of `AbstractDataFrame`s with
 `vcat`.
 
+The keyword argument `init` is the initial value to use in the reductions.
+It must be a data frame that has zero rows. It is not included when specifying
+`source` keyword argument and when determining metadata of the produced data frame.
```

Comment:
I don't understand what `not included when specifying `source` keyword argument` means. Since it has zero rows, isn't it obvious that it has no effect on the `source` column?

Reply:
The issue is:

> It can be a `Symbol` or an `AbstractString`, in which case the identifier will be the number of the passed source data frame, or a `Pair` consisting of a `Symbol` or an `AbstractString` and of a vector specifying the data frame identifiers (which do not have to be unique).

Which means that normally we count empty data frame even if we drop it. Here is an example:
```
julia> reduce(vcat, [DataFrame(), DataFrame(a=1), DataFrame(), DataFrame(a=2)], source="aaa")
2×2 DataFrame
 Row │ a      aaa
     │ Int64  Int64
─────┼──────────────
   1 │     1      2
   2 │     2      4
```
Note that `:aaa` column has indices `2` and `4` as data frames 1 and 3 had zero rows.

What the docstring says is that the data frame passed as `init` is not counted (as if it were counted indexing would start from 2 and not from 1).

Reply:
Got it. Then how about something like this?
```suggestion
It must be a data frame that has zero rows. It is not taken into account when
computing the value of the `source` column nor when determining metadata
of the produced data frame.
```